tests/assembly/asm/aarch64-types.rs

```diff
@@ -255,6 +268,20 @@ check!(vreg_i32x2 i32x2 vreg "fmov" "s");
 // CHECK: //NO_APP
 check!(vreg_i64x1 i64x1 vreg "fmov" "s");
 
+// neon-LABEL: vreg_f16x4:
+// neon: @APP
+// neon: vmov.f64 d{{[0-9]+}}, d{{[0-9]+}}
+// neon: @NO_APP
+#[cfg(neon)]
+check!(vreg_f16x4 f16x4 vreg "vmov.f64");
+
+// neon-LABEL: vreg_f16x8:
+// neon: @APP
+// neon: vorr q{{[0-9]+}}, q{{[0-9]+}}, q{{[0-9]+}}
+// neon: @NO_APP
+#[cfg(neon)]
+check!(vreg_f16x8 f16x8 vreg "vmov");
```

Comment:
I don't think these `neon` labels are actually getting tested, you need to add `neon` in the `//@ revisions:` list at the top and make sure `--cfg neon` gets passed somewhere (like https://github.com/rust-lang/rust/blob/249595384b4589c71a00d622c4253f62c7f5ddbd/tests/assembly/asm/arm-types.rs#L1-L7)

I'm not sure what combinations it makes sense to test, there's also a `fp16` target feature. Cc @workingjubilee since you did some work on neon and flags